.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,55 @@
+# This code is part of Qiskit.
```

Comment:
This workflow definitions replaces cron-prod.yml and cron-staging.yml that were mostly the same and included duplicated job definitions for different environments.

*Notable changes*
- avoid duplicated job definitions by using Github environments (see e.g. `legacy-production`)
- reduced the matrix options so that the integration tests are only run using the latest supported Python and ubuntu version
   - version specific issues are expected to be uncovered by unit tests
   - helps with stability and decreases test runtime because we're not "performance testing" the APIs with parallel test jobs
- left out the migration of cron-slow.yml and cron-slow-terra-main.yml for now
   - I believe the primary goal should be to stabilize the main path (integration tests) and then reenable slow tests  

Let's discuss objections related to these changes in this thread.